tests/test_rest_client.py

```
@@ -0,0 +1,260 @@
+    assert client._mfa_url.startswith("https://verify")
+
+
+@pytest.mark.asyncio
+async def test_authn_check_missing_status(config_na, session):
+    """Test _authn_check raises AuthenticationError if status missing."""
+    client = RESTClient(config_na, session)
+    session.post.return_value.__aenter__.return_value.json = AsyncMock(return_value={})
+    session.post.return_value.__aenter__.return_value.headers = {}
+    session.post.return_value.__aenter__.return_value.status = 200
+    with patch.object(RESTClient, "_resmed_response_error_check", new=AsyncMock()):
+        with pytest.raises(AuthenticationError):
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)